sections/performance/block-loop.md

```diff
@@ -0,0 +1,73 @@
+# Don't block the event loop
+
+<br/><br/>
+
+Node handles the Event Loop mostly on a single thread rotating through multiple queues. Operations with high complexity, large json parsing, unsafe regex querieies, and large IO operations are some of the operations that can cause the Event Loop to stall. Avoid this off-loading CPU intensive tasks to a dedicated service (e.g. job server), or breaking long tasks into small steps then using the Worker Pool are some examples of how to avoid blocking the Event Loop.
+
+## Example
+Let's take a look at an example from [Node Clinic](https://clinicjs.org/documentation/doctor/05-fixing-event-loop-problem).
+```
+const restify = require('restify')
+const server = restify.createServer()
+
+function sleep (ms) {
+  const future = Date.now() + ms
+  while (Date.now() < future);
+}
+
+server.get('/', function (req, res, next) {
+  sleep(30)
+  res.send({})
+  next()
+})
+
+server.listen(3000)
+```
+
+And when we benchmark this app, we start to see the latency caused by the long
+while loop.
+
+### Run the benchmark 
+`clinic doctor --on-port 'autocannon localhost:$PORT' -- node slow-event-loop`
+
+### The results
+
+```
+─────────┬────────┬────────┬────────┬────────┬───────────┬──────────┬───────────┐
+│ Stat    │ 2.5%   │ 50%    │ 97.5%  │ 99%    │ Avg       │ Stdev    │ Max       │
```

Comment:
This is a lot of information - what is the bottom line? what takeaway do we want the user to extract from this?

Reply:
Basically this is out the output from node clinic showing the high latency, but I can see it not being helpful without the context show node clinic show the loop block issue.

Reply:
@i0natan Do you have a suggestion on this section?